src/new-contract-interaction/new-contract-interaction.js

```diff
@@ -0,0 +1,163 @@
+const {
+  FindingType,
+  FindingSeverity,
+  Finding,
+  getJsonRpcUrl,
+} = require('forta-agent');
+const ethers = require('ethers');
+const axios = require('axios');
+
+// load configuration data from agent config file
+const {
+  newContractInteraction,
+  aaveEverestId: AAVE_EVEREST_ID,
+} = require('../../agent-config.json');
+
+// read the .env file and populate process.env with keys/values
+require('dotenv').config();
+
+const DEBUG = false; // enable/disable console logging for debugging
+
+// print log messages only if debug=true
+function log(message) {
+  if (DEBUG) {
+    console.log(message); // eslint-disable-line no-console
+  }
+}
+
+// computes contract age in days
+// units for currentTime and creationTime are SECONDS
+function getContractAge(currentTime, creationTime) {
+  return Math.floor((currentTime - creationTime) / 60 / (60 * 24));
+}
+
+// load config files
+const contractAddresses = require('../../contract-addresses.json');
+const tokenAddresses = require('./token-addresses.json');
```

Comment:
The workflow looks like the following:
- `mainnet.json` is committed to the repository
- A user manually runs `extract_addresses.js` to read in the content of 'mainnet.json`, modify it, then saved the result to `token-addresses.json`
- `token-addresses.json` is also committed to the repository
- The handler reads in `token-addresses.json` to perform its function

Therefore, the handler never runs `extract_addresses.js` and never otherwise interacts with `mainnet.json`.  Perhaps the Javascript file `extract_addresses.js` and the JSON file `mainnet.json` should not be committed as part of the handler?

Or is the intention that these files provide a blueprint/pattern/example for how other Scan Agent handler developers could work with the data?

Reply:
Yes, extract_addresses.js and mainnet.json are not part of the handler, but are related to setup of the handler.  We could automate the address extraction from mainnet.json, but adding the dependency on an external URL is not as safe as your other suggestion to query the ProtocolDataProvider contract.  I'm going to resolve this conversation but leave your other suggestion open.